src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                std::string flowMode = UtilityRoutines::MakeUPPERCase(fields.at("flow_mode").get<std::string>());
+                if (flowMode == "NOTMODULATED") {
+                    thisPLHP.flowMode = DataPlant::FlowMode::NotModulated;
+                } else if (flowMode == "CONSTANTFLOW") {
+                    thisPLHP.flowMode = DataPlant::FlowMode::Constant;
+                } else if (flowMode == "LEAVINGSETPOINTMODULATED") {
+                    thisPLHP.flowMode = DataPlant::FlowMode::LeavingSetpointModulated;
+                } else {
+                    ShowSevereError(state, fmt::format("{}{}=\"{}\"", RoutineName, cCurrentModuleObject, thisPLHP.name));
+                    ShowContinueError(state, "Invalid Flow Mode =" + flowMode);
+                    ShowContinueError(state, "Available choices are ConstantFlow, NotModulated, or LeavingSetpointModulated");
+                    ShowContinueError(state, "Flow mode NotModulated is assumed and the simulation continues.");
+                    // assume variable flow if not specified
+                    thisPLHP.flowMode = DataPlant::FlowMode::NotModulated; // default NotModulated
+                }
+
+                // if (fields.find("reference_coefficient_of_performance") != fields.end()) {
+                //    thisPLHP.referenceCOP = fields.at("reference_coefficient_of_performance").get<Real64>();
+                //} else {
+                //    Real64 defaultVal = 0.0;
+                //    if (!state.dataInputProcessing->inputProcessor->getDefaultValue(
+                //            state, cCurrentModuleObject, "reference_coefficient_of_performance", defaultVal)) {
+                //        // this error condition would mean that someone broke the input dictionary, not their
+                //        // input file.  I can't really unit test it so I'll leave it here as a severe error
+                //        // but excluding it from coverage
+                //        ShowSevereError(state,                                                                  // LCOV_EXCL_LINE
+                //                        "EIR PLHP: Reference COP not entered and could not get default value"); // LCOV_EXCL_LINE
+                //        errorsFound = true;                                                                     // LCOV_EXCL_LINE
+                //    } else {
+                //        thisPLHP.referenceCOP = defaultVal;
+                //    }
+                //}
+
+                // A9 outdoor_air_temperature_curve_input_variable
+                std::string oaTempCurveInputVar =
+                    UtilityRoutines::MakeUPPERCase(fields.at("outdoor_air_temperature_curve_input_variable").get<std::string>());
+                thisPLHP.oaTempCurveInputVar = static_cast<OATempCurveVar>(getEnumerationValue(OATempCurveVarNamesUC, oaTempCurveInputVar));
+                if (thisPLHP.oaTempCurveInputVar == OATempCurveVar::Invalid) {
```

Comment:
If this is a required field, then epJSON will trap this and you will never get here.

Reply:
Thanks! This field (A9) and the next field (A10) are indeed required field (and also with defaults). So followed the comment suggestion here and commented out the extra check code blocks/lines.